saltproc/openmc_depcode.py

```diff
@@ -124,6 +257,125 @@ def read_depleted_materials(self, read_at_end=False):
+                depleted_materials[name].vol = volume
+
+                if read_at_end:
+                    sp0 = openmc.StatePoint(self.output_path / 'openmc_simulation_n0.h5')
+                    heavy_metal_mass = starting_material.fissionable_mass * 1e-3
+                    #power = self._get_power_from_tallies(sp0, results[1].source_rate) * 1e-6
+                    power = results[1].source_rate * 1e-6
+                    days = results[1].time[1] / _SECONDS_PER_DAY
+                    burnup = power * days / heavy_metal_mass
+                else:
+                    burnup = 0
+                depleted_materials[name].burnup = burnup
+        del openmc_materials, depleted_openmc_materials, starting_openmc_materials
+        return depleted_materials
+
+    def _create_mass_percents_dictionary(self, mat, percent_type='ao'):
+        """Creates a dicitonary with nuclide codes
+        in zzaaam formate as keys, and material composition
+        in mass percent as values
+
+        Parameters
+        ----------
+        mat : openmc.Material
+            A material
+        percent_type : str
+            Percent type of material
+
+        Returns
+        -------
+            mass_dict : dict of int to float
+        """
+        percents = np.zeros(len(mat.nuclides))
+        nucs = []
+        at_mass = np.zeros(len(mat.nuclides))
+        for i, (nuc, pt, tp) in enumerate(mat.nuclides):
+            nucs.append(nuc)
+            percents[i] = pt
+            at_mass[i] = atomic_mass(nuc)
+
+        if percent_type == 'ao':
+            mass_percents = percents*at_mass / np.dot(percents, at_mass)
+        else:
+            mass_percents = percents
+        pyne_nucs = list(map(self._convert_nucname_to_pyne, nucs))
+
+        return dict(zip(pyne_nucs, mass_percents))
+
+    def _convert_nucname_to_pyne(self, nucname):
+        """Helper function for :func:`_create_mass_percents_dictionary`.
+        Converts an OpenMC-formatted nuclide name into a PyNE-formatted
+        nuclide name.
+
+        Parameters
+        ----------
+        nucname : str
+
+        Returns
+        -------
+        pyne_nucname : str
+
+        """
+        if nucname[-3:-1] == '_m':
+            nucname = nucname.split('_')[0]
+            nucname += 'M'
+        return nucname
+
+    def _get_power_from_tallies(self, sp, P):
+        # CLEANUP: import constants from openmc
+        fission_energy = sp.get_tally(name='fission_energy').mean.flatten()[0] # eV / src
+        heating = sp.get_tally(name='heating').mean.flatten()[0] # eV / src
+        Hp = JOULE_PER_EV * heating # J / src
+        f = P / Hp # src / s
+        power = fission_energy * f * JOULE_PER_EV # J / s
+        return power
+
+    def convert_nuclide_code_to_name(self, nuc):
+        nucname = pyname.name(nuc)
+        if nucname[-1] == 'M':
+            nucname = nucname[:-1]
+            nucname += '_m1'
+        return nucname
```

Comment:
Why is this one so much simpler than the serpent version?

Reply:
It has to do with ZAID conventions. Since OpenMC uses nuclide names instead of ZAID codes, it avoids all of that messy processing. I have [an issue](https://github.com/arfc/saltproc/issues/191) set up to address this in the future,.